.github/workflows/preview_ci.yml

```diff
@@ -1,69 +1,77 @@
 name: Preview - CI
 on:
-  push:
-    paths-ignore:
-      - '**/*.md'
-      - 'mkdocs.yml'
-      - 'src/docs/**/*'
-    branches: [ main ]
+  workflow_dispatch:
+  schedule:
+    # 4:19 AM UTC every day. A random time to avoid peak times of GitHub Actions.
+    - cron: '19 4 * * *'
 env:
   DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
   DOTNET_CLI_TELEMETRY_OPTOUT: true
 jobs:
   test:
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
     name: Build, Test, Deploy
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ubuntu-latest, windows-latest]
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
+      with:
+        # Necessary to count commits below. We assume that no more than 50 commits are pushed in a day to main.
+        fetch-depth: 50
+    - name: Check for new commits
+      id: check-commits
+      shell: pwsh
+      run: |
+        $commitCount = (git log --oneline --since '24 hours ago' | Measure-Object -Line).Lines
+        Write-Output "Count of new commits found in the last 24 hours: $commitCount."
+        "commit-count=$commitCount" >> $Env:GITHUB_OUTPUT
     - uses: actions/setup-node@v3
+      if: steps.check-commits.outputs.commit-count > 0
```

Comment:
Isn't there a way to just stop the workflow altogether in one step?

Reply:
Not in an analogous way to C#'s `return`. This is the only way to skip steps conditionally, use a single job, and make the workflow succeed even if the steps were skipped (which I think should happen; the run was successful, even if we didn't need to publish).

Alternatives:

- Having a separate job that produces a true/false, and then make the second job depend on that. This is what [Miek did](https://github.com/OrchardCMS/OrchardCore/pull/15551/files#diff-ed5e008a31c6bb458063a4792a9f71eba25bde81f1754b958c7bb4924cd838b6R16). The downside of this is that this check is a separate VM instance just for the script.
- Canceling the workflow, like [here](https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/actions/cancel-workflow/action.yml). That would show the workflow as canceled when the condition wasn't met instead of succeeding.